pennylane/math/single_dispatch.py

```diff
@@ -357,14 +357,20 @@ def _scatter_element_add_torch(tensor, index, value):
 # -------------------------------- JAX --------------------------------- #
 
 
+def _to_numpy_jax(x):
+    from jaxlib.xla_extension import DeviceArray
+
+    return np.array(x) if isinstance(x, DeviceArray) else x

```

Comment:
what's going on here? Don't we always want np arrays?

Reply:
Oh I think you just gave me an idea!!

Reply:
It worked!!!!!!!!!!

So, JAX has multiple types of arrays. On the forward pass, everything is a `DeviceArray`; this can be converted into a NumPy array via `np.array(x)`.

On the backward pass, however, JAX instead passes `ConcreteArray` objects, which don't support NumPy conversion. I suddenly wondered if they have a private attribute that allows you to extract the NumPy array, and they do!